### ISSUE TITLE
Error message for mis-configured apple sign in not different from device version wrong message

### ISSUE THREAD

**Issue by mikehardy** [invertase/react-native-apple-authentication#108]
> Very interesting - forgive me while I "active listen" here - I am going to try to say what you said, to see if I understand it
> 
> If you have not done the full iOS Developer Console setup for the app to enable apple sign in, then when you go to use it you get an error message about not having the right iOS version
> 
> Is that right? I am not 100% sure on the first part, but I am pretty sure I understand you've found the thing that causes a iOS version not supported message when really it has nothing to do with a version. Hopefully we can fix it

Yes, it is my point! 
Just to compare, in Google API Login, if we do mis configuration in Google Console (sha, key, or other thinks) we have code 10 error. But, if we trying use in unsupported device we have code 320. So, we can figure out where we do it wrong.
I suggestion to open a issue about that just to not lose this improve.

_Originally posted by @Lucasark in https://github.com/invertase/react-native-apple-authentication/issues/103#issuecomment-655132739_

**Comment by mikehardy** [comment]
@lucasark - thanks for helping diagnose this, we haven't known about this issue until now and couldn't get enough specifics to make anything actionable. I think we have something to work with here!

**Comment by mikehardy** [comment]
At minimum this needs a documentation fix so I tagged that, but also if the underlying Apple SDK gives us different error messages to separate these two cases at all, the code should differentiate them of course

**Comment by karanwadhwa** [comment]
I got the same error as well, the issue with me was `RNAppleAuthentication` did not get auto linked. I dont think this is an issue with the library, it's probably my project - I've had similar auto linking failures on this project before; nvm I ran the link command, installed the `RNAppleAuthentication` pod and it worked fine after.

**Comment by mikehardy** [comment]
Yeah - apparently @dawntommm just discovered in #185  that if you forget to pod install it maps to "AppleAuth is not supported on this device" also (or at least similarly) which is a pretty terrible user experience compared to other modules I actually maintain which will tell you the native module is missing :sweat_smile: 

Haven't had time to dig and fix this one yet but if anyone else does a PR would be lovely :pray:
